python_modules/dagster/dagster/_core/definitions/asset_key.py

```diff
@@ -196,6 +203,28 @@ def from_user_string(user_string: str) -> "AssetCheckKey":
         asset_key_str, name = user_string.split(":")
         return AssetCheckKey(AssetKey.from_user_string(asset_key_str), name)
 
+    @staticmethod
+    def from_db_string(db_string: str) -> Optional["AssetCheckKey"]:
+        try:
+            values = seven.json.loads(db_string)
+            if isinstance(values, dict) and values.keys() == {"asset_key", "name"}:
+                return AssetCheckKey(
+                    asset_key=check.not_none(AssetKey.from_db_string(values["asset_key"])),
+                    name=check.inst(values["name"], str),
+                )
+            else:
+                return None
+        except seven.JSONDecodeError:
+            return None
+
+    def to_db_string(self) -> str:
+        return seven.json.dumps({"asset_key": self.asset_key.to_string(), "name": self.name})
```

Comment:
`check_name` maybe incase we want to distinguish from some other future sibling to checks

Reply:
thoughts on that vs adding a `{"type": "check"}` in there?

it really feels like we're dangerously close to just re-inventing whitelist_for_serdes here, although this certainly has stricter backcompat requirements, so it does feel like we still need to keep it separate

Reply:
hmm, to me most terse and acceptable would be `{"check": name, "asset_key": path}` which is what i would bias towards. I think the impl complexity is effectively the same amongst the choices so I pull towards having less bytes to compare

Reply:
+1 preference for `check` or `check_name` over `name`